lib/src/functions/map.dart

```diff
@@ -86,6 +94,19 @@ final _deepMerge = _function("deep-merge", r"$map1, $map2", (arguments) {
   return _deepMergeImpl(map1, map2);
 });
 
+final _deepRemove =
+    _function("deep-remove", r"$map, $key, $keys...", (arguments) {
+  var map = arguments[0].assertMap("map");
+  var keys = [arguments[1], ...arguments[2].asList];
+  return _modify(map, keys.sublist(0, keys.length - 1), (value) {
```

Comment:
Should this call `_modify` with `overwrite = false`?

Reply:
I'm wondering if `_modify()` should handle this automatically by allowing the callback to return `null` to indicate that nothing should be changed.

Reply:
Good call, that's much cleaner.

Reply:
Although, would that require us to recurse to the leaf node in all cases, even if we know that intermediate maps won't be necessary?

Reply:
Once you reach the point where you don't have an actual map, there's nothing more to recurse through anyway—you can immediately call `modify(null)`, and choose whether to create nested maps or just exit early based on the result.